dss/config.py

```diff
@@ -12,6 +12,30 @@
 from .hcablobstore.gs import GSHCABlobStore
 
 
+class DeploymentStageMeta(EnumMeta):
+    _MAGIC_PREFIX = "IS_"
+
+    def __getattr__(cls, item: str):
+        if item.startswith(DeploymentStageMeta._MAGIC_PREFIX):
+            trailer = item[len(DeploymentStageMeta._MAGIC_PREFIX):]
+            attr = getattr(DeploymentStage, trailer, None)
+            if isinstance(attr, DeploymentStage):
+                return lambda: os.environ["DSS_DEPLOYMENT_STAGE"] == attr.value
+        raise AttributeError(item)
+
+
+class DeploymentStage(Enum, metaclass=DeploymentStageMeta):
+    """
+    Represents the current deployment stage.  Through the `DeploymentStageMeta` metaclass, we provide the magic methods
+    IS_{STAGE}, which return True if the current deployment stage matches `STAGE`.
+
+    e.g., IS_PROD() will return True iff the current deployment is a production deployment.
+    """
+    PROD = "prod"
+    STAGING = "staging"
+    DEV = "dev"
```

Comment:
Why use a metaclass when you can just use getattr to dynamically generate the method? That seems a bit simpler.

Reply:
__getattr__ does not work as a classmethod. This works like DeploymentStage.IS_PROD()

Reply:
ah I see. OK LGTM then.